prometheus_alerting_consistency.md

```diff
@@ -0,0 +1,104 @@
+    namespace label.
+* Alerts SHOULD include a cluster identifier label, especially in cases where alerts are defined or routed centrally for multiple services & clusters.
+  * This can hint to an SRE where the affected service is located. It can also be useful for aggregation & filtering when analysing groups of firing alerts.
+* All alerts SHOULD include an annotation (e.g. `runbook_url`) which directs the responder(s) to the location of the runbook (sometimes called a Standard Operating Procedure or SOP) that is specific to fixing that problem.
+
+## Critical Alerts
+
+Critical alerts are for alerting current and impending disaster situations.  These alerts page an SRE.  The situation should warrant waking someone in the middle of the night.
+
+Reserve critical level alerts only for reporting conditions that may lead to service unavailability.
+Failures of individual components should not trigger critical level alerts in general, unless they would result in that condition.
+Configure critical level alerts so they fire before the situation becomes irrecoverable.
+
+Example critical alert:
+
+```yaml
+- alert: BackendServiceStuck
+  expr: max_over_time(backend_state[5m]) != 1
+  for: 10m
+  labels:
+    severity: critical
+  annotations:
+    summary: 'The backend service is stuck in a non-ready state'
+    description: 'The backend service {{ $labels.name }} in the {{ $labels.namespace }} namespace, managed by operator {{ $labels.pod }} has been in a non-ready state for 10 minutes'
+    runbook_url: 'https://example.com/backend_service_stuck.asciidoc'
+```
+
+This alert fires if a backend service has *not* been ready for the last 10 minutes.
+This is a clear example of a critical issue that represents a threat to the operability of the service, and likely warrants paging someone.
+The alert has a clear summary and description annotations, and it links to a runbook with information on investigating and resolving the issue.
+
+The group of critical alerts should be small, very well-defined, highly documented, polished and with a high bar set for entry.
+
+## Warning Alerts
+
+The vast majority of alerts should use this severity.
+Issues at the warning level should be addressed in a timely manner, but don't pose an immediate threat to the operation of the service as a whole.
+If your alert does not meet the criteria in "Critical Alerts" above, it belongs to the warning level.
+
+Use warning level alerts for reporting conditions that may lead to inability to deliver individual features of the service, but not the service as a whole.
+Most alerts are likely to be warnings.
+Configure warning level alerts so that they do not fire until components have sufficient time to try to recover from the interruption automatically.
+You should have a policy for triaging warning alerts and responding in a timely manner e.g. issues created automatically on a backlog that is being actively worked from.
```

Comment:
The text we end up with here should try to match what teams already do (or plan to do asap).  Here's another draft ADR:  https://docs.google.com/document/d/1j2F8HgqYGsTnoq1fii2ZMGacuC6jYUp0wsDOVDXL1Tc @dofinn

Reply:
Agreed.
I've requested access to that doc to see how well these align.
I'll defer to @dofinn if there's something to align better in the mean time.

Reply:
this looks pretty good to me :) i find this such a subjective issue, but this is a good happy medium